bin/publish

```diff
@@ -19,14 +19,10 @@ readonly TAGS=(
   "latest"
 )
 
-# fetching tags is required for git_description to work
-git fetch --tags
-git_description=$(git describe)
-
 # if it’s not a tagged commit, VERSION will have extra characters (i.e. -g666c4b2), so that commit won't be published
 # only when tag matches the VERSION, push VERSION and latest releases
 # and x and x.y releases
-if [ "$git_description" = "v${VERSION}" ]; then
+if is_tagged_commit then
```

Comment:
```suggestion
if [[ ! -z "${TAG_NAME:-}" ]]; then
```

Reply:
Hey @izgeri, 
I do understand your suggestion, but do you think that this is aligned with Jenkins's best practice there - 
"The Jenkinsfile should contain minimal logic, it should only be used for assigning scripts to nodes and collecting results. The bulk of the build/test/release logic should be contained in other scripts in the repo." ?

Reply:
can you please explain how this does not align with that best practice?

the Jenkinsfile only contains the logic about _when_ steps should be run (eg on tags)

the rest of the logic is entirely in the scripts (like this one we are commenting on, `bin/publish`)

this is consistent with the best practices recommended by the infra team for our pipelines, and with the direction all of our other projects are moving. in addition, it addresses a known bug where sometimes pipelines wouldn't publish because new commits have been added to master since the tag.